pinot-common/src/main/java/org/apache/pinot/common/restlet/resources/SegmentErrorInfo.java

```diff
@@ -0,0 +1,94 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.restlet.resources;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.TimeZone;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+
+/**
+ * This class is used to represent errors related to a segment.
+ */
+@SuppressWarnings("unused")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SegmentErrorInfo {
+  private static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss z";
+
+  private final String _timeStamp;
+  private final String _error;
+  private final String _exception;
+
+  /**
+   *
+   * @param exception Exception object
+   * @param errorMessage Error Message
+   * @param timeStampInMillisSinceEpoch Timestamp of the error
+   */
+  public SegmentErrorInfo(Exception exception, String errorMessage, long timeStampInMillisSinceEpoch) {
+    _timeStamp = epochToSDF(timeStampInMillisSinceEpoch);
+    _error = errorMessage;
+    _exception = (exception != null) ? ExceptionUtils.getStackTrace(exception) : null;
+  }
+
+  /**
+   * This constructor is specifically for JSON ser/de.
+   *
+   * @param exception Exception stack trace
+   * @param error Error message
+   * @param timeStamp Time stamp of the error in Simple Date Format.
+   *
+   */
+  @JsonCreator
+  public SegmentErrorInfo(@JsonProperty("exception") String exception, @JsonProperty("error") String error,
+      @JsonProperty("timeStamp") String timeStamp) {
+    _timeStamp = timeStamp;
+    _error = error;
+    _exception = exception;
+  }
+
+  public String getTimeStamp() {
+    return _timeStamp;
+  }
+
+  public String getError() {
+    return _error;
+  }
+
+  public String getException() {
+    return _exception;
+  }
+
+  /**
+   * Utility function to convert epoch in millis to SDF of form "yyyy-MM-dd HH:mm:ss z".
+   *
+   * @param millisSinceEpoch Time in millis to convert
+   * @return SDF equivalent
+   */
+  private static String epochToSDF(long millisSinceEpoch) {
+    SimpleDateFormat sdf = new SimpleDateFormat(DATE_FORMAT, Locale.getDefault());
```

Comment:
We should put the SDF as a constant without initializing the format for every call